### PR TITLE
Add sidebar nav

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Add navigation--sidebar pattern

## QA

- Pull code
- Run `./run serve --watch`
- View http://0.0.0.0:8101/vanilla-framework/examples/patterns/navigation/sidebar/
- Check docs make sense

## Details

Fixes https://www.github.com/vanilla-framework/vanilla-docs-theme/issues/45

### CODE REVIEW

File: examples/patterns/navigation/sidebar.html [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  </ul>
+  <hr>
+  <ul class="p-list">
+    <li><a class="p-link--external" href="#">Supplementary link 1</a></li>
+    <li><a class="p-link--external" href="#">Supplementary link 2</a></li>
+  </ul>
+</div>
+
+<script>
+  let navSidebarLinks = document.querySelectorAll(".sidebar__first-level a");
```
Comment (anthonydillon) [inline examples/patterns/navigation/sidebar.html]: We should be targeting `js-` classes here.

File: scss/_patterns_navigation.scss [hunk 2]
```diff
@@ -34,11 +35,98 @@
...
-  $background-color: $color-brand,
+@mixin navigation-sidebar {
+  .p-navigation--sidebar {
+    @include navigation-pattern(
+      $background-color: $color-x-light,
+      $border-color: $color-mid-light,
+      $text-color: $color-dark
+    );
+
+    a {
```
Comment (anthonydillon) [inline scss/_patterns_navigation.scss]: I feel it would be better to target a class within patterns in case we have a link that we don't want to be styled as a side navigation item.

**Comment by yaili** [comment]
Can you experiment with the menu drop down to see if it can scroll on mobile?

**Comment by kwm14** [comment]
Looking good :+1:

File: examples/patterns/navigation/sidebar.html [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illum ducimus maiores, sunt sequi officia neque sit? Error blanditiis modi mollitia, reprehenderit id illum! Quas suscipit officia vitae voluptatum dolores, facere?</p>
+
+<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illum ducimus maiores, sunt sequi officia neque sit? Error blanditiis modi mollitia, reprehenderit id illum! Quas suscipit officia vitae voluptatum dolores, facere?</p>
+
+<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illum ducimus maiores, sunt sequi officia neque sit? Error blanditiis modi mollitia, reprehenderit id illum! Quas suscipit officia vitae voluptatum dolores, facere?</p>
+
+<script>
+  // Sidebar menu overlay toggle
+  let toggleElements = document.querySelectorAll('[data-js="sidebar-toggle"]');
```
Comment (anthonydillon) [inline examples/patterns/navigation/sidebar.html]: We should hook javascript to classes as detailed in the practices: https://github.com/canonical-webteam/practices/blob/master/coding/stylesheets.md#javascript-namespacing

If you believe targeting data attributes is better then please create a proposal on the practises document first.

**Comment by barrymcgee** [comment]
@anthonydillon Added proposal to webteam practices: https://github.com/canonical-webteam/practices/issues/31

**Comment by barrymcgee** [comment]
@anthonydillon Ready for re-review.

**Comment by anthonydillon** [comment]
## QA issue
The menu items `p-icon--menu` should have cursor: pointer if they are clickable. This pattern is also missing a few aria attributes.

File: scss/_patterns_icons.scss [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    &--x-large {
+      @include vf-icon-size(2.25rem);
+    }
+
+    &--xx-large {
+      @include vf-icon-size($sp-xxx-large);
+    }
+  }
```
Comment (richmccartney) [inline scss/_patterns_icons.scss]: Hey @barrymcgee can you add in this snippet of SCSS too, then we don't need to land another PR in dashboard theme.

```
// Style ovrides for icons within a button 
 [class*="p-button-"] {
  [class*="p-icon-"] {
    top: -1px;
    vertical-align: middle;
  }
}
```

This just resolves a small issue for alignment of icons with inside of a button. I kept the styles to icons as the button patterns shouldn't infer these changes.

**Comment by richmccartney** [comment]
@barrymcgee apart from my one comment and the conflicts this looks good to me 👍
